Site/dataobjects/SiteApiCredential.php

```diff
@@ -74,11 +74,38 @@ public function loadByApiKey($key, SiteInstance $instance =  null)
 			return false;
 
 		$this->initFromRow($row);
+
 		$this->generatePropertyHashes();
 
 		return true;
 	}
 
+	// }}}
+	// {{{ public function getPromotionByType()
+
+	public function getPromotionByType($type)
+	{
+		$sql = sprintf(
+			'SELECT *
+			FROM %s
+			INNER JOIN Promotion ON
+				Promotion.api_credential = %s.id
+			WHERE %s.id = %d
+				AND Promotion.api_sign_on_type = %s',
+			$this->table,
+			$this->table,
+			$this->table,
+			$this->id,
+			$this->db->quote($type, 'text')
+		);
+
+		return SwatDB::query(
```

Comment:
This returns a set of promotions but the method suggests it returns a singular promotion. Either update the method to be plural or call `getFirst()` on the result.